code/modules/surgery/repair_puncture.dm

```diff
@@ -0,0 +1,111 @@
+	surgery.operated_bodypart.receive_damage(brute=rand(4,8), sharpness=SHARP_EDGED, wound_bonus = 10)
+
+///// Sealing the vessels back together
+/datum/surgery_step/seal_veins
+	name = "weld veins" // if your doctor says they're going to weld your blood vessels back together, you're either A) on SS13, or B) in grave mortal peril
+	implements = list(TOOL_CAUTERY = 100, /obj/item/gun/energy/laser = 90, TOOL_WELDER = 70, /obj/item = 30)
+	time = 40
+
+/datum/surgery_step/seal_veins/tool_check(mob/user, obj/item/tool)
+	if(implement_type == TOOL_WELDER || implement_type == /obj/item)
+		return tool.get_temperature()
+
+	return TRUE
+
+/datum/surgery_step/seal_veins/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/datum/wound/pierce/pierce_wound = surgery.operated_wound
+	if(pierce_wound)
+		display_results(user, target, "<span class='notice'>You begin to meld some of the split blood vessels in [target]'s [parse_zone(user.zone_selected)]...</span>",
+			"<span class='notice'>[user] begins to meld some of the split blood vessels in [target]'s [parse_zone(user.zone_selected)] with [tool].</span>",
+			"<span class='notice'>[user] begins to meld some of the split blood vessels in [target]'s [parse_zone(user.zone_selected)].</span>")
+	else
+		user.visible_message("<span class='notice'>[user] looks for [target]'s [parse_zone(user.zone_selected)].</span>", "<span class='notice'>You look for [target]'s [parse_zone(user.zone_selected)]...</span>")
+
+/datum/surgery_step/seal_veins/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	var/datum/wound/pierce/pierce_wound = surgery.operated_wound
+	if(pierce_wound)
+		display_results(user, target, "<span class='notice'>You successfully meld some of the split blood vessels in [target]'s [parse_zone(target_zone)] with [tool].</span>",
+			"<span class='notice'>[user] successfully melds some of the split blood vessels in [target]'s [parse_zone(target_zone)] with [tool]!</span>",
+			"<span class='notice'>[user] successfully melds some of the split blood vessels in [target]'s [parse_zone(target_zone)]!</span>")
+		log_combat(user, target, "dressed burns in", addition="INTENT: [uppertext(user.a_intent)]")
+		pierce_wound.blood_flow -= 0.5
+		if(pierce_wound.blood_flow > 0)
+			surgery.status = REALIGN_INNARDS
+			to_chat(user, "<span class='notice'><i>There still seems to be misaligned blood vessels to finish...<i></span>")
+		else
+			to_chat(user, "<span class='green'>You've repaired all the internal damage in [target]'s [parse_zone(target_zone)]!</span>")
+	else
+		to_chat(user, "<span class='warning'>[target] has no puncture there!</span>")
+	return ..()
+
+/datum/surgery_step/seal_veins/failure(mob/user, mob/living/target, target_zone, obj/item/tool, datum/surgery/surgery, var/fail_prob = 0)
+	..()
+	if(istype(tool, /obj/item/stack))
```

Comment:
why is this checking for/consuming from a stack of items? none of the items that you can use for this surgery are stacks

Reply:
carryover psata